### PR TITLE
drivers: clock_control: stm32h7 use a macro to define VCO_INPUT_RANGE

### PR DESCRIPTION
**Pull request by xhpohanka**
When HSI is used as pll clock source we must use also its divider

### CODE REVIEW

File: drivers/clock_control/clock_stm32_ll_h7.c [hunk 1]
```diff
@@ -545,7 +545,7 @@ static int stm32_clock_control_init(struct device *dev)
 	/* Main PLL configuration and activation */
 	LL_RCC_PLL_SetSource(LL_RCC_PLLSOURCE_HSI);
 
-	pllsrc_clock = HSI_VALUE;
+	pllsrc_clock = HSI_VALUE / CONFIG_CLOCK_STM32_HSI_DIVISOR;
```
Comment (erwango) [inline drivers/clock_control/clock_stm32_ll_h7.c:548]: This is correct but, I wonder if we can just remove `pllsrc_clock` and replace by use of constant `PLLSRC_FREQ` which is already defined conditionally:
```
/* Choose PLL SRC */
#ifdef CONFIG_CLOCK_STM32_PLL_SRC_HSI
#define PLLSRC_FREQ  ((HSI_FREQ)/(CONFIG_CLOCK_STM32_HSI_DIVISOR))
#elif defined(CONFIG_CLOCK_STM32_PLL_SRC_CSI)
#define PLLSRC_FREQ  CSI_FREQ
#elif defined(CONFIG_CLOCK_STM32_PLL_SRC_HSE)
#define PLLSRC_FREQ  HSE_FREQ
#else
#define PLLSRC_FREQ 0
#endif
```

**Review by erwango — COMMENTED** [review]
Fix is correct and welcome.

File: drivers/clock_control/clock_stm32_ll_h7.c (hunk 1 above)
Comment (xhpohanka) [reply drivers/clock_control/clock_stm32_ll_h7.c:548]: It makes a perfect sense. I will update the PR.

Comment (xhpohanka) [reply drivers/clock_control/clock_stm32_ll_h7.c:548]: Maybe even whole `get_vco_input_range()` function should be changed to macro and warn about wrong settings in compile time...

Comment (erwango) [reply drivers/clock_control/clock_stm32_ll_h7.c:548]: go ahead!

Comment (lochej) [reply drivers/clock_control/clock_stm32_ll_h7.c:548]: You're right, that makes sense I missed that when making the original PR. Using the constants will be best !

Comment (lochej) [reply drivers/clock_control/clock_stm32_ll_h7.c:538]: Then you can remove any ```pllsrc_clock``` occurance has it's no longer needed. I added it only to  compute the VCO input range.

Comment (xhpohanka) [reply drivers/clock_control/clock_stm32_ll_h7.c]: > Then you can remove any pllsrc_clock occurance
I thought I have done it, thanks for comment. Fixed now

File: drivers/clock_control/clock_stm32_ll_h7.c [hunk 2]
```diff
@@ -62,6 +62,19 @@
 #define PLLSRC_FREQ 0
 #endif
 
+#define VCO_FREQ	(PLLSRC_FREQ / CONFIG_CLOCK_STM32_PLL_M_DIVISOR)
+#if (1000000UL <= VCO_FREQ && VCO_FREQ <= 2000000UL)
```
Comment (erwango) [inline drivers/clock_control/clock_stm32_ll_h7.c:66]: Nit: Break a line here for readability:
```suggestion
#define VCO_FREQ	(PLLSRC_FREQ / CONFIG_CLOCK_STM32_PLL_M_DIVISOR)

#if (1000000UL <= VCO_FREQ && VCO_FREQ <= 2000000UL)
```

**Review by lochej — APPROVED** [review]
LGTM

**Review by erwango — CHANGES_REQUESTED** [review]

File: drivers/clock_control/clock_stm32_ll_h7.c [hunk 3]
```diff
@@ -62,6 +62,19 @@
...
+#define VCO_INPUT_RANGE LL_RCC_PLLINPUTRANGE_1_2
+#elif (2000000UL < VCO_INPUT_RANGE && VCO_INPUT_RANGE <= 4000000UL)
```
Comment (xhpohanka) [inline drivers/clock_control/clock_stm32_ll_h7.c:68]: And noone noticed this mistake from copying :)

File: drivers/clock_control/clock_stm32_ll_h7.c (hunk 2 above)
Comment (erwango) [reply drivers/clock_control/clock_stm32_ll_h7.c]: Now it it more readable I can see it ;-)

File: drivers/clock_control/clock_stm32_ll_h7.c (hunk 3 above)
Comment (erwango) [reply drivers/clock_control/clock_stm32_ll_h7.c]: Now it it more readable, I can see it ;-)

Comment (lochej) [reply drivers/clock_control/clock_stm32_ll_h7.c]: Exactly that's a good one ! 🙃 I had a attentive look at the constants but didn't catch that one. I could have seen it by testing though ^^

File: drivers/clock_control/clock_stm32_ll_h7.c [hunk 4]
```diff
@@ -62,6 +62,21 @@
...
+#if (1000000UL <= VCO_FREQ && VCO_FREQ <= 2000000UL)
+#define VCO_INPUT_RANGE LL_RCC_PLLINPUTRANGE_1_2
+#elif (2000000UL < VCO_FREQ && VCO_FREQ <= 4000000UL)
+#define VCO_INPUT_RANGE LL_RCC_PLLINPUTRANGE_2_4
+#elif (4000000UL < VCO_FREQ && VCO_FREQ <= 8000000UL)
+#define VCO_INPUT_RANGE LL_RCC_PLLINPUTRANGE_4_8
+#elif (8000000UL < VCO_FREQ && VCO_FREQ <= 16000000UL)
+#define VCO_INPUT_RANGE LL_RCC_PLLINPUTRANGE_8_16
+#else
+#pragma message STRINGIFY(VCO_FREQ)
```
Comment (erwango) [inline drivers/clock_control/clock_stm32_ll_h7.c:76]: Tryong to get into the error, I got:
```
/local/mcu/zephyrproject/zephyr/drivers/clock_control/clock_stm32_ll_h7.c:76:9: note: #pragma message: (8000000 / 63)
   76 | #pragma message STRINGIFY(VCO_FREQ)
      |         ^~~~~~~
/local/mcu/zephyrproject/zephyr/drivers/clock_control/clock_stm32_ll_h7.c:77:2: error: #error "PLL1 VCO frequency input range out of range " STRINGIFY(VCO_FREQ)
   77 | #error "PLL1 VCO frequency input range out of range " STRINGIFY(VCO_FREQ)
      |  ^~~~~
```
I guess this is not the expected result...

Comment (xhpohanka) [reply drivers/clock_control/clock_stm32_ll_h7.c:76]: Damn, sorry, I'm doing to many things at once, this should not be there
BTW is there a way how to print out result of macro inside `#pragma` or `#error`

Comment (lochej) [reply drivers/clock_control/clock_stm32_ll_h7.c]: @xhpohanka There is no way, preprocessor doesn't compute the result, it's the compiler's job. But you can still do math in #if statements.

**Review by erwango — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**